packages/pluggableWidgets/accordion-web/src/Accordion.editorConfig.ts

```diff
@@ -105,3 +109,120 @@ export function check(values: AccordionPreviewProps): Problem[] {
 
     return errors;
 }
+
+export function getPreview(values: AccordionPreviewProps): StructurePreviewProps | null {
+    return {
+        type: "Container",
+        borders: true,
+        children: values.groups
```

Comment:
This whole function is kind of extremely difficult to follow 😅  It's long, a lot of things are going through each other, and there are also array function happening. Will add some suggestion

Reply:
I agree. I already had that feeling a bit before creating the PR. Refactored the code according to your comments.